tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -1171,6 +1171,23 @@ void test_gh_5253() {
     g_regexTester.should_not_match("a", "()*");
 }
 
+void test_gh_5364() {
+    // GH-5364 `<regex>`: Allow initial ] to start character ranges in basic regular expressions
+    for (syntax_option_type option : {basic, extended, grep, egrep}) {
+        g_regexTester.should_match("]", "[]-_]", option);
+        g_regexTester.should_match("^", "[]-_]", option);
+        g_regexTester.should_match("_", "[]-_]", option);
+        g_regexTester.should_not_match("-", "[]-_]", option);
+
+        g_regexTester.should_match("]", "[]a]", option);
+        g_regexTester.should_match("a", "[]a]", option);
+        g_regexTester.should_not_match("a]", "[]a]", option);
+        g_regexTester.should_not_match("]a", "[]a]", option);

```

Comment:
We should also test that it doesn't match a single character like `_` (which is in the range of `]` to `a`, but here they are being used individually).